monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ResultsFolderController.java

```diff
@@ -49,24 +51,46 @@
     @RolesAllowed(SecurityRoles.VIEWER)
     @RequestMapping(method = RequestMethod.GET, produces = "application/json")
     @ResponseBody
-    public List<Map<String, String>> resultsFolderJson(@PathVariable("tenant") final String tenant) {
-        final TenantContext context = _tenantContextFactory.getContext(tenant);
+    public List<Map<String, String>> resultsFolderJson(@PathVariable("tenant") final String tenant,
+            @RequestParam(value = "not_before", required = false) final String timestamp) {
 
+        final TenantContext context = _tenantContextFactory.getContext(tenant);
         final RepositoryFolder resultsFolder = context.getResultFolder();
-
         final List<Map<String, String>> result = new ArrayList<>();
 
         {
             final List<RepositoryFile> files =
                     resultsFolder.getFiles(null, FileFilters.ANALYSIS_RESULT_SER.getExtension());
             for (final RepositoryFile file : files) {
                 final Map<String, String> map = new HashMap<>();
-                map.put("filename", file.getName());
+                final String name = file.getName();
+                if (!isEligibleJob(timestamp, name)) {
+                    continue;
+                }
+                map.put("filename", name);
                 map.put("repository_path", file.getQualifiedPath());
+
                 result.add(map);
             }
         }
 
         return result;
     }
+
+    public boolean isEligibleJob(final String timestamp, final String name) {
+        final Date searchedTimestamp;
+        if (timestamp == null || Long.valueOf(timestamp) < 0) {
+            // we do not care about the timestamp, therefore we add the job to the result
+            return true;
+        } else {
+            searchedTimestamp = new Date(Long.valueOf(timestamp));
+        }
+        // get the timestamp of the job
+        final String timestampString = name.substring(name.lastIndexOf("-") + 1, name.indexOf("."));
+        final Date jobDate = new Date(Long.valueOf(timestampString));
+        if (jobDate.after(searchedTimestamp)) {
```

Comment:
Not important at all, but you can just return `jobDate.after(searchedTimestamp)` directly.